Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Extensions/ApiGatewayResponseExtensions.cs

```diff
@@ -0,0 +1,273 @@
+            { "Connection", "keep-alive" }
+        };
+
+        switch (emulatorMode)
+        {
+            case ApiGatewayEmulatorMode.Rest:
+                headers.Add("x-amzn-RequestId", Guid.NewGuid().ToString("D"));
+                headers.Add("x-amz-apigw-id", GenerateRequestId());
+                headers.Add("X-Amzn-Trace-Id", GenerateTraceId());
+                break;
+            case ApiGatewayEmulatorMode.HttpV1:
+            case ApiGatewayEmulatorMode.HttpV2:
+                headers.Add("Apigw-Requestid", GenerateRequestId());
+                break;
+        }
+
+        return headers;
+    }
+
+    /// <summary>
+    /// Generates a random X-Amzn-Trace-Id for REST API mode.
+    /// </summary>
+    /// <returns>A string representing a random X-Amzn-Trace-Id in the format used by API Gateway for REST APIs.</returns>
+    /// <remarks>
+    /// The generated trace ID includes:
+    /// - A root ID with a timestamp and two partial GUIDs
+    /// - A parent ID
+    /// - A sampling decision (always set to 0 in this implementation)
+    /// - A lineage identifier
+    /// </remarks>
+    private static string GenerateTraceId()
+    {
+        var timestamp = DateTimeOffset.UtcNow.ToUnixTimeSeconds().ToString("x");
+        var guid1 = Guid.NewGuid().ToString("N");
+        var guid2 = Guid.NewGuid().ToString("N");
+        return $"Root=1-{timestamp}-{guid1.Substring(0, 12)}{guid2.Substring(0, 12)};Parent={Guid.NewGuid().ToString("N").Substring(0, 16)};Sampled=0;Lineage=1:{Guid.NewGuid().ToString("N").Substring(0, 8)}:0";
+    }
+
+    /// <summary>
+    /// Generates a random API Gateway request ID for HTTP API v1 and v2.
+    /// </summary>
+    /// <returns>A string representing a random request ID in the format used by API Gateway for HTTP APIs.</returns>
+    /// <remarks>
+    /// The generated ID is a 15-character string consisting of lowercase letters and numbers, followed by an equals sign.
+    /// </remarks>
+    private static string GenerateRequestId()
+    {
+        return Guid.NewGuid().ToString("N").Substring(0, 8) + Guid.NewGuid().ToString("N").Substring(0, 7) + "=";
+    }
+
+    /// <summary>
+    /// Sets the response body on the <see cref="HttpResponse"/>.
+    /// </summary>
+    /// <param name="response">The <see cref="HttpResponse"/> to set the body on.</param>
+    /// <param name="body">The body content.</param>
+    /// <param name="isBase64Encoded">Whether the body is Base64 encoded.</param>
+    /// <param name="apiGatewayEmulator">The <see cref="ApiGatewayEmulatorMode"/> being used.</param>
+    private static void SetResponseBody(HttpResponse response, string? body, bool isBase64Encoded, ApiGatewayEmulatorMode apiGatewayEmulator)
+    {
+        if (!string.IsNullOrEmpty(body))
+        {
+            byte[] bodyBytes;
+            if (isBase64Encoded && ApiGatewayEmulatorMode.Rest != apiGatewayEmulator) // rest api gateway doesnt automatically decode the response
```

Comment:
If `isBase64Encoded` is `true` we should always decode the body. I think what you are seeing is that in API Gateway you need to configure your REST API for binary media. And then it will decode it. From our emulator's point of view we should assume that they have correctly configured binary media and so we should decode base 64 encoded bodies.

In CloudFormation you can configure binary media by adding the following resource:
```
    "TheApi": {
      "Type": "AWS::Serverless::Api",
      "Properties": {
        "StageName": "Prod",
        "BinaryMediaTypes": [
          "*~1*"
        ]
      }
    },
```

Then on the event configuration for the serviceless function reference the API:
```
        ],
        "PackageType": "Zip",
        "Events": {
          "RootGet": {
            "Type": "Api",
            "Properties": {
              "Path": "/",
              "Method": "GET",
              "RestApiId" : { "Ref" : "TheApi" }
            }
          }
``

Reply:
i see. that certainly makes more sense. ill test it out to see what happens

Reply:
yeah i just tested this and you are right. i will update this logic  and add a test case

Reply:
ive fixed this now and have also added an integration and unit test